src/test/java/com/example/demo/mapper/AnimeMapperTest.java

```diff
@@ -56,7 +56,8 @@ public class AnimeMapperTest {
     @DataSet(value = "anime.yml")
     @ExpectedDataSet(value = "expectedAfterInsertAnime.yml", ignoreCols = "id")
     void アニメが登録できること() {
-        animeMapper.createAnime(new Anime(3, "Gintama", "Comedy"));
+        animeMapper.createAnime(new Anime("Gintama", "Comedy"));
+        animeMapper.createAnime(new Anime("Your Name", "Romantic Fantasy"));
```

Comment:
createAnimeにわたした引数のAnimeオブジェクトにIDが設定されることを確認してほしいです。
ですので
```suggestion
        Anime anime = new Anime("Your Name", "Romantic Fantasy")
        animeMapper.createAnime(anime);
        // animeに新しいIDが設定されていることを確認する
```
というようにコードを記述しないといけません。

Reply:
すみませんが、IDは以下のように確認したらいいでしょうか。
`assertThat(anime.getId()).isNotNull();`

Reply:
登録実行前後でidが変わっていることを確認すればいいと思います。
最初は0のはずですが実行後には新しいidになっていることが期待値です。

Reply:
テストが実行する時にIDが増えてきて、それをどう確認したらいいですか。

Reply:
下記のように`createAnime`実行前後でidが更新されていることを確認すればいいと思います。
```java
        Anime anime = new Anime("Your Name", "Romantic Fantasy")
        // animeのidがデフォルト値であること
        animeMapper.createAnime(anime);
        // animeのidが更新されていること
```

Reply:
あ、でもAnime.javaのidはIntegerですからティリさんのいうように、
```
createAnime実行前: `assertThat(anime.getId()).isNull();`
createAnime実行後: `assertThat(anime.getId()).isNotNull();`
```
でもよいですね。

Reply:
承知しました。ありがとうございます。

Reply:
修正しました。ご確認お願いします。
7c8e90f8814faad37c42ad7a0e1179547b3bb2c2

Reply:
いいですね。
念のため、auto_incrementを確認するためにgintamaAnimeとyourAnimeのidを比較しておきましょう。
あとから登録した方がidの値が大きいはずですね。

Reply:
gintamaAnimeとyourAnimeのidを比べるケースを追加しました。ご確認お願いします。
1fd3688a35b6b448505d408f6c452a53a82cce27

Reply:
問題ないです！